tests/nvidiagpu/gpu_suite_test.go

```diff
@@ -27,4 +34,29 @@ func TestGPUDeploy(t *testing.T) {
 var _ = JustAfterEach(func() {
 	reporter.ReportIfFailed(
 		CurrentSpecReport(), currentFile, tsparams.ReporterNamespacesToDump, tsparams.ReporterCRDsToDump, clients.SetScheme)
+
+	dumpDir := inittools.GeneralConfig.GetDumpFailedTestReportLocation(currentFile)
+	if dumpDir != "" {
```

Comment:
I think we want also 
```go
import (
    "github.com/onsi/ginkgo/v2/types"
)
```
...
```go
	if !types.SpecStateFailureStates.Is(CurrentSpecReport().State) {
		return
	}
```

to collect must-gather only in case of failure, just like `reporter.ReportIfFailed()`. 
Even better, we can create a reusable function to run a must-gather script if it was provided. E.g. `reporter.MustGatherIfFailed()`.
Let's address it in a new PR.